index.bs

```diff
@@ -1167,6 +1167,30 @@ which might be expressed as a bitmask in another language,
 use a dictionary object instead.
 This object can be passed around as easily as a single bitmask value.
 
+<h3 id="properties-vs-methods">Properties vs. Methods</h3>
+
+Sometimes it is unclear whether to use an property or a method.
+
+  * Property getters must not have any (observable) side effects.
+    If you have expected side effects, use a method.
+  * Property getters should not perform any expensive operations.
+    If a getter requires performing an expensive operation, it should be a method.
+    (A notable failure of the platform in this regard is getters like offsetTop performing layout; do not repeat this mistake.)
```

Comment:
Like I wrote, I'm not sure about this point, performance should not influence API shape ideally, since performance concerns are often transient and APIs are much more long-lived. However the last point about `offsetTop` might be covered by the previous point about side effects?

Reply:
offsetTop is a combination of both sins.

How does this look? https://github.com/w3ctag/design-principles/pull/262/files#r633975504

Reply:
Much better!